java-util/src/main/java/io/druid/java/util/common/CompressionUtils.java

```diff
@@ -252,6 +255,21 @@ public InputStream openStream() throws IOException
     return result;
   }
 
+  public static void validateZipOutputFile(String sourceFilename, final File outFile, final File outDir) throws IOException
+  {
+    // check for evil zip exploit that allows writing output to arbitrary directories
+    final File canonicalOutFile = outFile.getCanonicalFile();
+    final String canonicalOutDir = outDir.getCanonicalPath();
+    if (!canonicalOutFile.toPath().startsWith(canonicalOutDir)) {
```

Comment:
This check should be stricter. It would be okay with canonicalOutFile = `/tmpp/x` and canonicalOutDir = `/tmp`, but it shouldn't be.

Reply:
I think this is fine, following check returns `false`:

```
    File file = new File("/tmpp/x").getCanonicalFile();
    String dir = new File("/tmp").getCanonicalPath();
    System.out.println("result: " + file.toPath().startsWith(dir));
...
result: false

```

Reply:
I think the test you pasted only passes by accident. On my machine,

```
scala> new File("/tmp").getCanonicalPath()
res2: String = /private/tmp

scala> new File("/tmpp/x").getCanonicalFile().toPath()
res3: java.nio.file.Path = /tmpp/x
```

So if `/tmp` were not actually in `/private/tmp` then the result would have been `true`.

Reply:
I guess `/tmp` on osx is a bad example, this also returns false:

```
    File file = new File("/nottmppp/x").getCanonicalFile();
    String dir = new File("/nottmp").getCanonicalPath();
    System.out.println("result: " + file.toPath().startsWith(dir));
```

Reply:
Ah okay I see what is going on. "startsWith" isn't just checking the strings, it's doing something smarter. Okay that is cool then.